google-cloud-firestore/lib/google/cloud/firestore/batch.rb

```diff
@@ -0,0 +1,1013 @@
+        #
+        #     # Get/run a query
+        #     b.get(query).each do |city|
+        #       puts "#{city.document_id} has #{city[:population]} residents."
+        #     end
+        #   end
+        #
+        def get obj
+          ensure_not_closed!
+          ensure_service!
+
+          obj = coalesce_get_argument obj
+
+          if obj.is_a?(Document::Reference)
+            doc = get_all([obj]).first
+            yield doc if block_given?
+            return doc
+          end
+
+          return enum_for(:get, obj) unless block_given?
+
+          results = service.run_query obj.parent_path, obj.grpc
+          results.each do |result|
+            next if result.document.nil?
+            yield Document.from_query_result(result, self)
+          end
+        end
+        alias_method :run, :get
+
+        ##
+        # Create a document with the provided object values.
+        #
+        # The batch will fail if the document already exists.
+        #
+        # @param [String, Document::Reference] doc A string representing the
+        #   path of the document, or a document reference object.
+        # @param [Hash] data The document's fields and values.
+        #
+        # @example Create a document using a document path:
+        #   require "google/cloud/firestore"
+        #
+        #   firestore = Google::Cloud::Firestore.new
+        #
+        #   firestore.batch do |b|
+        #     # Create a document
+        #     b.create("cities/NYC", { name: "New York City" })
+        #   end
+        #
+        # @example Create a document using a document reference:
+        #   require "google/cloud/firestore"
+        #
+        #   firestore = Google::Cloud::Firestore.new
+        #
+        #   firestore.batch do |b|
+        #     # Get a document reference
+        #     nyc_ref = b.doc "cities/NYC"
+        #
+        #     # Create a document
+        #     b.create(nyc_ref, { name: "New York City" })
+        #   end
+        #
+        def create doc, data
+          ensure_not_closed!
+
+          doc_path = coalesce_doc_path_argument doc
+
+          Convert.create_writes(doc_path, data).each do |write|
+            @writes << write
+          end
+
+          nil
+        end
+
+        ##
+        # Write to document with the provided object values. If the document
+        # does not exist, it will be created. By default, the provided data
+        # overwrites existing data, but the provided data can be merged into the
+        # existing document using the `merge` argument.
+        #
+        # @param [String, Document::Reference] doc A string representing the
+        #   path of the document, or a document reference object.
+        # @param [Hash] data The document's fields and values.
+        # @param [true, String|Symbol, Array<String|Symbol>] merge When provided
+        #   and `true` all data is merged with the existing docuemnt data.  When
```

Comment:
s/docuemnt/document/ (in 12 locations total)

Reply:
Fixed.